paddle/fluid/operators/math/blas.cc

```diff
@@ -13,10 +13,47 @@
 // limitations under the License.
 
 #include "paddle/fluid/operators/math/blas.h"
+
+#include <utility>
 namespace paddle {
 namespace operators {
 namespace math {
-// Do nothing. Blas is a header only library.
+MatDim GetMatDim(const framework::DDim& dim, int num_flatten_cols, bool trans) {
```

Comment:
The name `GetMatDim` is pretty vague. And the optional flattening and transposing operations make users hard to grasp the idea of this function. Can we decompose this function into:

- FlattenColums
- Transpose
- Collapse

Reply:
By learning from [cudnn APIs](https://docs.nvidia.com/deeplearning/sdk/cudnn-developer-guide/index.html#cudnnCreateTensorDescriptor), I rename this API to `CreateMatDescriptor`.